addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +120,134 @@ def _get_duration_from_tracking(self, trackings):
+
+
+        Also consider overriding _get_rotting_depends_fields() and _get_rotting_domain().
+
+        Certain views have access to widgets to display rotting status:
+            'rotting' for kanbans, 'rotting_statusbar_duration' for forms, 'badge_rotting' for lists.
+
+        :return: bool: whether the rotting feature has been configured for this model
+        """
+        return 'rotting_threshold_days' in self[self._track_duration_field] and 'date_last_stage_update' in self and (
+            not self  # api.model call
+            or any(stage.rotting_threshold_days for stage in self[self._track_duration_field])
+        )
+
+    def _get_rotting_depends_fields(self):
+        """
+        fields added to this method through override should likely also be returned by _get_rotting_domain() override
+
+        :return: the array of fields that can affect the ability of a resource to rot
+        """
+        if hasattr(self, '_track_duration_field') and 'rotting_threshold_days' in self[self._track_duration_field]:
+            return ['date_last_stage_update', f'{self._track_duration_field}.rotting_threshold_days']
+        return []
+
+    def _get_rotting_domain(self):
+        """
+        fields added to this method through override should likely also be returned by _get_rotting_depends_fields() override
+
+        :return: domain: conditions that must be met so that the field can be considered rotting
+        """
+        return [
+            (f'{self._track_duration_field}.rotting_threshold_days', '!=', 0),
+        ]
+
+    @api.depends(lambda self: self._get_rotting_depends_fields())
+    def _compute_rotting(self):
+        """
+        A resource is rotting if its stage has not been updated in a number of days depending on its
+        stage's rotting_threshold_days value, assuming it matches _get_rotting_domain() conditions.
+
+        If the rotting_threshold_days field is not defined on the tracked module,
+        or if the value of rotting_threshold_days is 0,
+        then the resource will never rot.
+        """
+        if not self._is_rotting_feature_enabled():
+            self.update({
+                'is_rotting': False,
+                'rotting_days': 0,
+            })
+            return
+        now = self.env.cr.now()
+        rot_enabled = self.filtered_domain(self._get_rotting_domain())
+        others = self - rot_enabled
+        for stage, records in rot_enabled.grouped(self._track_duration_field).items():
+            rotting = records.filtered(lambda record: record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days) < now)
+            for record in rotting:
+                record.is_rotting = True
+                record.rotting_days = (now - record.date_last_stage_update).days
+            others += records - rotting
+        others.update({
+            'is_rotting': False,
+            'rotting_days': 0,
+        })
```

Comment:
Is it really worth it to call update when we need to manually build the dict for it?
Can't we have 2 simple assignations instead?